src/components/pages/UserPage.js

```diff
@@ -34,16 +41,23 @@ export const UserPage = ({ match }) => {
       }),
     );
 
+    // TODO: de-couple creating chart from getting data
     const tempD3Data = Object.entries(tempData).map(([key, value]) => {
       return { name: key, value: value.length };
     });
     setD3Data(tempD3Data);
-    setRepos(repoData);
+
+    // clone previous userRepos and append new key value pair
+    const newUserRepos = Object.assign({}, userRepos);
+    Object.assign(newUserRepos, { [user]: repoData });
+    setUserRepos(newUserRepos);
   };
 
   useEffect(() => {
-    getData();
-    return () => {};
+    // only getData if its not already populated
+    if (!repos.length) {
```

Comment:
I had to remove this if condition when testing fetching a second user. or does that work now?

Reply:
it works now